### DISCUSSION TITLE
how to connect the traefik file provider to k3s?

### DISCUSSION THREAD

**Discussion by PahaCore** [k3s-io/k3s#7320] [General]
```yaml
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: traefik
  namespace: kube-system
spec:
  valuesContent: |-
    additionalArguments:
      - --accesslog=true
      - --global.checknewversion
      - --entryPoints.vpn.address=:51820/udp
      - --entrypoints.web.http.redirections.entryPoint.to=:443
      - --entrypoints.web.http.redirections.entryPoint.scheme=https
    volumeMounts:
      - mountPath: /home
        name: file
  volumes:
    - name: file
      hostPath:
        path: /home/paha/deployment/file/
        type: Directory
  providers:
    file:
      directory: /home
```yaml

**Comment by brandond** [comment]
This isn't the best place to ask questions about traefik. We just package the chart; we don't maintain the chart or Traefik itself. If you have questions about how to use it, I would probably go to https://github.com/traefik/traefik-helm-chart.

I can tell you that just pasting random bits of pod spec into the HelmChartConfig spec isn't going to do what you want, though. You should look at the chart values and see if what you want is even possible. If it is possible, then you would want to provide the desired configuration in the ValuesContent section of the HelmChartConfig.

https://github.com/traefik/traefik-helm-chart/blob/master/traefik/values.yaml
